src/Dialect/ONNX/ONNXDimAnalysis.cpp

```diff
@@ -361,11 +361,11 @@ static bool exploreSameDimsUsingShapeHelper(const DimAnalysis::DimT &dim,
   ONNXOpShapeHelper *shapeHelper =
       shape_op.getShapeHelper(op, {}, nullptr, nullptr);
   // If no shape helper, or unimplemented, just abort.
-  if (!shapeHelper || !shapeHelper->isImplemented())
+  if (!shapeHelper)
```

Comment:
Could you explain what's wrong with calling `shapeHelper->isImplemented()`?

Reply:
There is nothing wrong with it, but if the shape helper is not implemented we still need to delete it or we will leak it's memory. This happens a few lines down

Reply:
I got it! That's why we have to distinguish the two conditions. Thanks for the explanation!